ext/node/ops/vm_internal.rs

```diff
@@ -0,0 +1,595 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::v8;
+use deno_core::v8::MapFnTo;
+
+pub const PRIVATE_SYMBOL_NAME: v8::OneByteConst =
+  v8::String::create_external_onebyte_const(b"node:contextify:context");
+
+/// An unbounded script that can be run in a context.
+#[derive(Debug)]
+pub struct ContextifyScript {
+  script: v8::Global<v8::UnboundScript>,
+}
+
+impl ContextifyScript {
+  pub fn new(
+    scope: &mut v8::HandleScope,
+    source_str: v8::Local<v8::String>,
+  ) -> Result<Self, AnyError> {
+    let source = v8::script_compiler::Source::new(source_str, None);
+
+    let unbound_script = v8::script_compiler::compile_unbound_script(
+      scope,
+      source,
+      v8::script_compiler::CompileOptions::NoCompileOptions,
+      v8::script_compiler::NoCacheReason::NoReason,
+    )
+    .ok_or_else(|| type_error("Failed to compile script"))?;
+    let script = v8::Global::new(scope, unbound_script);
+    Ok(Self { script })
+  }
+
+  pub fn eval_machine<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+    _context: v8::Local<v8::Context>,
```

Comment:
So that means `Script.runInContext` doesn't work correctly. Can you add a TODO here and maybe add a failing test that we can then update?

Reply:
Also a TODO for options object would be good (or maybe stub it out now?)

Reply:
It works correctly, see `Script::run_in_context` in `vm.rs`. The context is not used here, it'll probably be needed when adding `options`.